### PR TITLE
Add microphone permission guideline in the migration guide

### PR DESCRIPTION
**Pull request by bobiechen-twilio**
Add a section in the migration guide for requesting microphone permission in the apps using 3.x SDK.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -533,6 +534,54 @@ An example of managing the `TVODefaultAudioDevice` while connecting a CallKit Ca
 
 See [CallKit Example](https://github.com/twilio/voice-quickstart-objc/blob/3.x/ObjCVoiceCallKitQuickstart/ViewController.m) for the complete implementation.
 
+#### <a name="microphone-permission"></a>Microphone Permission
+Microphone permission is not optional in the Voice iOS 2.X SDK. Permission must be granted by the user in addition to the usage description being presented in the plist file. Error will be raised when making or accepting calls without microphone permission granted. In the Voice iOS 3.X SDK will continue to connect the call even when the microphone permission is denied or disabled by the user. The application needs to check record permission prior to connecting a call and provide proper experience to the users.
+
+```
+- (void)makeCall {
```
Comment (idelgado) [inline README.md:541]: Add this logic to the quickstart code itself.

File: README.md [hunk 2]
```diff
@@ -533,6 +534,54 @@ An example of managing the `TVODefaultAudioDevice` while connecting a CallKit Ca
 
 See [CallKit Example](https://github.com/twilio/voice-quickstart-objc/blob/3.x/ObjCVoiceCallKitQuickstart/ViewController.m) for the complete implementation.
 
+#### <a name="microphone-permission"></a>Microphone Permission
+Microphone permission is not optional in the Voice iOS 2.X SDK. Permission must be granted by the user in addition to the usage description being presented in the plist file. Error will be raised when making or accepting calls without microphone permission granted. In the Voice iOS 3.X SDK will continue to connect the call even when the microphone permission is denied or disabled by the user. The application needs to check record permission prior to connecting a call and provide proper experience to the users.
```
Comment (idelgado) [inline README.md]: Lets rephase to the following 

> Unlike Voice iOS 2.X SDKs where microphone permission is not optional in Voice 3.X SDKs, the call will connect even when the microphone permission is denied or disabled by the user. To ensure the microphone permission is enabled prior to making or accepting a call you can add the following to request permission beforehand:

**Review by idelgado — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -533,6 +534,54 @@ An example of managing the `TVODefaultAudioDevice` while connecting a CallKit Ca
 
 See [CallKit Example](https://github.com/twilio/voice-quickstart-objc/blob/3.x/ObjCVoiceCallKitQuickstart/ViewController.m) for the complete implementation.
 
+#### <a name="microphone-permission"></a>Microphone Permission
+Unlike Voice iOS 2.X SDKs where microphone permission is not optional in Voice 3.X SDKs, the call will connect even when the microphone permission is denied or disabled by the user. To ensure the microphone permission is enabled prior to making or accepting a call you can add the following to request permission beforehand:
```
Comment (idelgado) [inline README.md]: > to request the permission beforehand:

**Review by idelgado — CHANGES_REQUESTED** [review]

**Review by idelgado — APPROVED** [review]
👍 with respect to docs. @piyushtank should review the snippet source and quickstart source.

File: ObjCVoiceQuickstart/ViewController.m [hunk 4]
```diff
@@ -75,17 +75,82 @@ - (IBAction)placeCall:(id)sender {
...
+    typeof(self) __weak weakSelf = self;
+    [self checkRecordPermission:^(BOOL permissionGranted) {
+        typeof(self) __strong strongSelf = weakSelf;
+        if (!permissionGranted) {
+            UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"Voice Quick Start"
+                                                                                     message:@"Microphone permission not granted."
+                                                                              preferredStyle:UIAlertControllerStyleAlert];
+
+            UIAlertAction *continueWithoutMic = [UIAlertAction actionWithTitle:@"Continue without microphone"
+                                                                         style:UIAlertActionStyleDefault handler:^(UIAlertAction *action) {
```
Comment (piyushtank) [inline ObjCVoiceQuickstart/ViewController.m]: new line at `handler:`
applies to other places also

File: ObjCVoiceQuickstart/ViewController.m [hunk 5]
```diff
@@ -75,17 +75,82 @@ - (IBAction)placeCall:(id)sender {
...
+            [strongSelf makeCall:strongSelf.outgoingValue.text];
         }];
         
         [self toggleUIState:NO showCallControl:NO];
         [self startSpin];
     }
 }
 
+- (void)makeCall:(NSString *)to {
+    TVOConnectOptions *connectOptions = [TVOConnectOptions optionsWithAccessToken:[self fetchAccessToken] block:^(TVOConnectOptionsBuilder *builder) {
```
Comment (piyushtank) [inline ObjCVoiceQuickstart/ViewController.m]: new line before `block:` for readability

File: README.md [hunk 6]
```diff
@@ -533,6 +534,54 @@ An example of managing the `TVODefaultAudioDevice` while connecting a CallKit Ca
 
 See [CallKit Example](https://github.com/twilio/voice-quickstart-objc/blob/3.x/ObjCVoiceCallKitQuickstart/ViewController.m) for the complete implementation.
 
+#### <a name="microphone-permission"></a>Microphone Permission
+Unlike Voice iOS 2.X SDKs where microphone permission is not optional in Voice 3.X SDKs, the call will connect even when the microphone permission is denied or disabled by the user. To ensure the microphone permission is enabled prior to making or accepting a call you can add the following to request the permission beforehand:
```
Comment (piyushtank) [inline README.md]: s/call/Call

suggestions:

Unlike Voice iOS 2.X SDKs where microphone permission is not optional in Voice 3.X SDKs, the Call will connect even when the microphone permission is denied or disabled by the user`, and the SDK will play the remote audio`. To ensure the microphone permission is enabled prior to making or accepting a call you can add the following to request the permission beforehand:

**Review by piyushtank — APPROVED** [review]
Looks good. left a minor comment.

File: README.md [hunk 7]
```diff
@@ -533,6 +534,54 @@ An example of managing the `TVODefaultAudioDevice` while connecting a CallKit Ca
 
 See [CallKit Example](https://github.com/twilio/voice-quickstart-objc/blob/3.x/ObjCVoiceCallKitQuickstart/ViewController.m) for the complete implementation.
 
+#### <a name="microphone-permission"></a>Microphone Permission
+Unlike Voice iOS 2.X SDKs where microphone permission is not optional in Voice 3.X SDKs, the call will connect even when the microphone permission is denied or disabled by the user, and the SDK will play the remote audio. To ensure the microphone permission is enabled prior to making or accepting a call you can add the following to request the permission beforehand:
```
Comment (piyushtank) [inline README.md:538]: s/call/Call

**Review by piyushtank — APPROVED** [review]

File: README.md (hunk 7 above)
Comment (bobiechen-twilio) [reply README.md:538]: Decided to go as is based on oral discussion.

**Pull request merged**